tests/test_generators/test_plantuml.py

```diff
@@ -45,114 +37,88 @@ class "Person" {
-        gen = PlantumlGenerator(SCHEMA)
-        plantuml = gen.serialize()
-        with open(PLANTUML_OUT, "w") as stream:
-            stream.write(plantuml)
-        self._in(MARKDOWN_HEADER, plantuml, "diagram header is missing")
-        self._in(MARKDOWN_FOOTER, plantuml, "diagram footer is missing")
-        self._in(PERSON, plantuml)
-        self._in(PERSON2MEDICALEVENT, plantuml)
-        self._in(FAMILIALRELATIONSHIP2PERSON, plantuml)
-
-    def test_serialize_direct(self):
-        """Test default serialization (structural), no markdown block"""
-        gen = PlantumlGenerator(SCHEMA)
-        plantuml = gen.serialize()
-        self.assertNotIn("```", plantuml, "markdown header should not be present")
-        self._in(PERSON, plantuml)
-        self._in(PERSON2MEDICALEVENT, plantuml)
-
-    def test_serialize_selected(self):
-        """Test serialization of selected elements"""
-        gen = PlantumlGenerator(SCHEMA)
-        plantuml = gen.serialize(classes=["FamilialRelationship"])
-        self.assertNotIn("MarriageEvent", plantuml, "MarriageEvent not reachable from selected")
-        self._in(FAMILIALRELATIONSHIP2PERSON, plantuml, "dangling references should be included")
-
-    def test_generate_svg(self):
-        gen = PlantumlGenerator(SCHEMA)
-        svg_temp_dir = tempfile.mkdtemp()
-        gen.serialize(directory=svg_temp_dir)
-        assert_svgfile_contains(
-            "KitchenSink.svg",
-            "<svg xmlns=",
-            outdir=svg_temp_dir,
-        )
-        assert_svgfile_contains(
-            "KitchenSink.svg",
-            '<!--class FakeClass--><g id="elem_FakeClass">',
-            outdir=svg_temp_dir,
-        )
-        assert_svgfile_contains(
-            "KitchenSink.svg",
-            ">FakeClass</text>",
-            outdir=svg_temp_dir,
-        )
-        assert_svgfile_contains(
-            "KitchenSink.svg",
-            "<line style=",
-            outdir=svg_temp_dir,
-        )
-        assert_svgfile_contains(
-            "KitchenSink.svg",
-            "</svg>",
-            outdir=svg_temp_dir,
-        )
-        shutil.rmtree(svg_temp_dir)
-
-    def _in(self, s1, s2, message: str = None):
-        self.assertIn(s1.replace(" ", ""), s2.replace(" ", ""), message)
-
-
-if __name__ == "__main__":
-    unittest.main()
+@pytest.mark.parametrize(
+    "input_class,expected",
+    [
+        # check that plantUML header and footer are present
+        # for one class, like `Person`, extensible to other classes
+        ("Person", MARKDOWN_HEADER),
+        ("Person", MARKDOWN_FOOTER),
+        # check that expected plantUML class diagram blocks are present
+        # when diagrams are generated for different classes
+        ("Person", PERSON),
+        ("Dataset", DATASET2PERSON),
+        ("MedicalEvent", PERSON2MEDICALEVENT),
+        ("FamilialRelationship", FAMILIALRELATIONSHIP2PERSON),
+    ],
+)
+def test_serialize_selected(input_class, expected, kitchen_sink_path):
```

Comment:
I'm not sure this is the best use of `parametrize`. Especially because this will end up re-generating for the `Person` class a few times. I think the intention of the test might be more clear without parameterizing and doing something like:

```
serialize Person class
assert ...
assert ...
assert ...

serialize Dataset class
assert ...

serialize MeticalEvent class
assert ...
```

Reply:
That's a great suggestion @pkalita-lbl 😁 will keep that in mind while using `parametrize`. For the sake of this test though, I actually think it's okay to remove the lines that create the class multiple times.